document/store/src/main/java/io/camunda/document/store/aws/AwsDocumentStore.java

```diff
@@ -193,6 +209,33 @@ private Either<DocumentError, DocumentLink> linkDocumentInternal(
     }
   }
 
+  private Either<DocumentError, Void> verifyContentHashInternal(
+      final String documentId, final String contentHashToVerify) {
+    try {
+      final HeadObjectResponse documentInfo = getDocumentInfo(documentId);
+      if (documentInfo == null) {
+        return Either.left(new DocumentNotFound(documentId));
+      }
+      if (!documentInfo.hasMetadata()) {
+        return Either.left(new DocumentError.InvalidInput("No metadata found for document"));
+      }
+
+      if (!documentInfo.metadata().containsKey(CONTENT_HASH_METADATA_KEY)) {
+        return Either.left(new DocumentError.InvalidInput("No content hash found for document"));
```

Comment:
@mo-okocha do we have unit tests for this?

Reply:
Unit tests have been added for each branch in the method